src/main/java/com/cloudbees/hudson/plugins/folder/computed/DefaultOrphanedItemStrategy.java

```diff
@@ -230,6 +253,46 @@ private static boolean disabled(TopLevelItem item) {
 
     @Override
     public <I extends TopLevelItem> Collection<I> orphanedItems(ComputedFolder<I> owner, Collection<I> orphaned, TaskListener listener) throws IOException, InterruptedException {
+        if (abortBuilds) {
+            List<Run<?, ?>> abortedBuilds = new ArrayList<>();
+            for (I item: orphaned) {
+                for (Job<?, ?> job: item.getAllJobs()) {
```

Comment:
what happens if this was not an organisation folder but a regular multibranch?

IIUC owner would be the Multibranch folder, and orphaned would be a WorkflowJob, so orphaned.getAllJobs() would be `null` or empty.

does this only work for an organisation folder when an entire repository is removed?

or is this handled somewhere else?

possibly I am missing something

Reply:
https://github.com/jenkinsci/jenkins/blob/1a807197ba0c574c6f9c79803fcd1d5dc14fbf43/core/src/main/java/hudson/model/Job.java#L506-L509 ?